src/hotspot/share/gc/g1/g1EvacFailureRegions.hpp

```diff
@@ -55,8 +57,15 @@ class G1EvacFailureRegions {
 
   bool contains(uint region_idx) const;
   void par_iterate(HeapRegionClosure* closure,
-                   HeapRegionClaimer* _hrclaimer,
+                   HeapRegionClaimer* hrclaimer,
                    uint worker_id) const;
+  void initialize_chunks(uint active_workers, const char* task_name);
+  // Iterate through all chunks in regions that failed evacuation during the entire collection.
+  void par_iterate_chunks_in_regions(G1HeapRegionChunkClosure* chunk_closure,
+                                     uint worker_id) const;
+
+  // Do necessary preparation for evacuation failure regions
+  void prepare_regions();
```

Comment:
I think the whole of `prepare_regions` should be integrated as a `G1BatchedTask`; from an initial look there does not seem to be anything in cleanup phase 1 that is dependent on what this code updates.
Simply add a method that returns an `G1AbstractSubTask` for use in the `G1BatchedGangTask` as done e.g. for `G1RemSet::create_cleanup_after_scan_heap_roots_task`.